test/localDate.test.ts

```diff
@@ -507,6 +507,206 @@ describe('A value object representing a local date', () => {
+        },
+        '2015-08-31 + P2Y': {
+            start: LocalDate.of(2015, 8, 31),
+            period: Period.ofYears(2),
+            expected: LocalDate.of(2017, 8, 31),
+        },
+        '2015-08-31 + P1Y2M3D': {
+            start: LocalDate.of(2015, 8, 31),
+            period: Period.of(1, 2, 3),
+            expected: LocalDate.of(2016, 11, 3),
+        },
+        '2016-11-03 + P-1Y-2M-': {
+            start: LocalDate.of(2016, 11, 3),
+            period: Period.of(-1, -2, -3),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+    }))('should add a period to a local date %s', (_, scenario) => {
+        const result = scenario.start.plus(scenario.period);
+
+        expect(result.toString()).toStrictEqual(scenario.expected.toString());
+    });
+
+    type SubtractPeriodScenario = AddPeriodScenario;
+
+    it.each(Object.entries<SubtractPeriodScenario>({
+        '2015-08-31 - P0D': {
+            start: LocalDate.of(2015, 8, 31),
+            period: Period.zero(),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2015-08-31 - P1D': {
+            start: LocalDate.of(2015, 9, 1),
+            period: Period.ofDays(1),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2015-08-31 - P2D': {
+            start: LocalDate.of(2015, 9, 2),
+            period: Period.ofDays(2),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2015-08-31 - P7D': {
+            start: LocalDate.of(2015, 9, 7),
+            period: Period.ofWeeks(1),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2015-08-31 - P14D': {
+            start: LocalDate.of(2015, 9, 14),
+            period: Period.ofWeeks(2),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2015-08-31 - P1M1D': {
+            start: LocalDate.of(2015, 10, 1),
+            period: Period.of(0, 1, 1),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2015-08-31 - P2M': {
+            start: LocalDate.of(2015, 10, 31),
+            period: Period.ofMonths(2),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2015-08-31 - P1Y': {
+            start: LocalDate.of(2016, 8, 31),
+            period: Period.ofYears(1),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2015-08-31 - P2Y': {
+            start: LocalDate.of(2017, 8, 31),
+            period: Period.ofYears(2),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2015-08-31 - P1Y2M3D': {
+            start: LocalDate.of(2016, 11, 3),
+            period: Period.of(1, 2, 3),
+            expected: LocalDate.of(2015, 8, 31),
+        },
+        '2016-11-03 - P-1Y-2M-': {
+            start: LocalDate.of(2015, 8, 31),
+            period: Period.of(-1, -2, -3),
+            expected: LocalDate.of(2016, 11, 3),
+        },
+    }))('should subtract a period from a local date %s', (_, scenario) => {
```

Comment:
```suggestion
    }))('should subtract $period from $start', (_, scenario) => {
```
It's better to keep the keys off; otherwise, we'll have the extra burden of maintaining it in sync.